server/src/main/java/org/opensearch/indices/replication/OngoingSegmentReplications.java

```diff
@@ -147,6 +148,16 @@ void startSegmentCopy(GetSegmentFilesRequest request, ActionListener<GetSegmentF
      */
     CopyState prepareForReplication(CheckpointInfoRequest request, FileChunkWriter fileChunkWriter) throws IOException {
         final CopyState copyState = getCachedCopyState(request.getCheckpoint());
+        if (copyState.getCheckpoint()
+            .getLatestSupportedCodec()
+            .equalsIgnoreCase(request.getCheckpoint().getLatestSupportedCodec()) == false) {
+            // cancel(request.getTargetAllocationId(), "Requested unsupported codec version");
+            logger.trace("Requested unsupported codec version {}", request.getCheckpoint().getLatestSupportedCodec());
+            throw new CancellableThreads.ExecutionCancelledException(
+                new ParameterizedMessage("Requested unsupported codec version {}", request.getCheckpoint().getLatestSupportedCodec())
+                    .toString()
+            );
```

Comment:
@Poojita-Raj : I think we need integration tests here verifying behavior 
1. Both nodes running on same OS version
2. Source node running on higher lucene codec
3. Target node running on higher lucene codec

Reply:
I am verifying 2 and 3 with the unit tests added in OngoingSegmentReplicationTests. Currently with this release, I'm unable to add in integration tests since it requires having this change in 2 different distributions of OS with differing codecs. But they can definitely be included in further releases.